core/chains/evm/client/errors.go

```diff
@@ -66,25 +59,49 @@ const (
 	OutOfCounters
 )
 
-type ClientErrors = map[int]*regexp.Regexp
+type ClientErrors struct {
+	errorMap map[int]*regexp.Regexp
+}
+
+func NewClientErrors(errors map[int]*regexp.Regexp) *ClientErrors {
+	return &ClientErrors{
+		errorMap: errors,
+	}
+}
+
+// ErrIs returns true if err matches any provided error types
+func (e *ClientErrors) ErrIs(err error, errorTypes ...int) bool {
```

Comment:
What about just dropping the `=` to make it a named type instead of an alias?
```suggestion
type ClientErrors map[int]*regexp.Regexp

// ErrIs returns true if err matches any provided error types
func (e ClientErrors) ErrIs(err error, errorTypes ...int) bool {
```

Reply:
Nice, sounds good!